typescript/infra/scripts/check/check-warp-deploy.ts

```diff
@@ -1,29 +1,101 @@
 import chalk from 'chalk';
+import { Gauge, Registry } from 'prom-client';
 
 import { WarpRouteIds } from '../../config/warp.js';
+import { submitMetrics } from '../../src/utils/metrics.js';
 import { Modules } from '../agent-utils.js';
 
-import { check, getCheckArgs } from './check-utils.js';
+import {
+  getCheckWarpDeployArgs,
+  getGovernor,
+  logViolations,
+} from './check-utils.js';
 
-async function checkWarp() {
-  const argv = await getCheckArgs().argv;
+async function main() {
+  const { environment, asDeployer, chain, fork, context, pushMetrics } =
+    await getCheckWarpDeployArgs().argv;
+
+  const metricsRegister = new Registry();
+  const checkerViolationsGauge = new Gauge({
```

Comment:
should this use `checkerViolationsGaugeObj`? or should it replace `checkerViolationsGaugeObj`

Reply:
Good spot